signer/migrations/0003__create_tables.sql

```diff
@@ -184,3 +191,111 @@ CREATE TABLE sbtc_signer.withdrawal_reject_events (
     created_at    TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
 );
 
+-- Represents a combined transaction package which is broadcasted to the Bitcoin
+-- network. The transaction package is built up of multiple transactions which
+-- are tracked separately in the `sweep_transactions` table. A transaction
+-- package may contain the sweeping transactions for both deposit and
+-- withdrawal requests.
+CREATE TABLE sbtc_signer.sweep_packages (
+    -- Internal ID of the package
+    id SERIAL PRIMARY KEY,
+    -- The Bitcoin block hash at which this package was created.
+    created_at_block_hash BYTEA NOT NULL,
+    -- The Bitcoin market fee rate at the time this package was created.
+    market_fee_rate DOUBLE PRECISION NOT NULL,
+    -- Timestamp of when this package was created.
+    created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+-- Represents an individual transaction within a broadcasted sweep transaction
+-- package. Individual deposit and withdrawal requests reference back to these
+-- transactions to keep track of both the overall transaction package as well as
+-- the individual Bitcoin transactions they are related to.
+CREATE TABLE sbtc_signer.sweep_transactions (
+    -- Internal ID of the transaction.
+    id BIGSERIAL PRIMARY KEY,
+    -- The ID of the grouping transaction package.
+    sweep_package_id INTEGER NOT NULL,
+    -- The Bitcoin transaction ID of the transaction.
+    txid BYTEA NOT NULL,
+    -- The signer UTXO being spent in this transaction.
+    utxo_txid BYTEA NOT NULL,
+    utxo_output_index INTEGER NOT NULL,
+    -- The total amount of the transaction.
+    amount BIGINT NOT NULL,
+    -- The fee paid for the transaction.
+    fee BIGINT NOT NULL,
+    -- The fee rate in satoshis per vByte used for this transaction.
+    fee_rate DOUBLE PRECISION NOT NULL,
+    -- The timestamp that the transaction was broadcast at. This should be
+    -- set after we know that the transaction was successfully broadcast.
+    is_broadcast BOOLEAN NOT NULL DEFAULT FALSE,
+
+    FOREIGN KEY (sweep_package_id)
+        REFERENCES sbtc_signer.sweep_packages(id)
+);
+
+-- Represents a single withdrawal request which has been included in a sweep
+-- transaction package. Withdrawal requests have a unique ID so we use that here
+-- to reference the withdrawal request together with its Stacks block hash,
+-- which can be retrieved from the `withdrawal_requests` table.
+CREATE TABLE sbtc_signer.swept_withdrawals (
+    -- Internal ID of the swept withdrawal.
+    id BIGSERIAL PRIMARY KEY,
+    -- References the `packaged_transaction` in which this withdrawal was
+    -- included.
+    sweep_transaction_id INTEGER NOT NULL,
```

Comment:
Why not use the transaction ID here?

Reply:
I think this is part of both the preceeding and succeeding comments - let's discuss further there :)

Reply:
Yeah let's do that. I'll set something up for Monday.

Reply:
Oh I just misread this. Yeah let's discuss further there.

Reply:
From our DB meeting, I'll revert this to natural keys.

Reply:
Natural keys in [25c1b2f](https://github.com/stacks-network/sbtc/pull/710/commits/25c1b2ffd730e593228cc57e12574671aa9530cd)